sympy/core/tests/test_assumptions.py

```diff
@@ -1062,3 +1062,10 @@ def test_issue_16313():
     assert (l*x).is_real is False
     assert (l*x*x).is_real is None  # since x*x can be a real number
     assert (-x).is_positive is False
+
+def test_issue_16579():
```

Comment:
Maybe add a comment here that complex may not imply infinite in the future.

Reply:
Okay, I would add comment.